openstack/pkg/designate/client.go

```diff
@@ -0,0 +1,213 @@
+package designate
+
+import (
+	"errors"
+	"fmt"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+	"github.com/gophercloud/gophercloud/openstack/dns/v2/recordsets"
+	"github.com/gophercloud/gophercloud/openstack/dns/v2/zones"
+	"github.com/libdns/libdns"
+	"os"
+	"time"
+)
+
+func (p *Provider) getRecords(recordSets []recordsets.RecordSet) ([]libdns.Record, error) {
+	var records []libdns.Record
+	for _, j := range recordSets {
+		tmp := libdns.Record{
+			ID:   j.ID,
+			Type: j.Type,
+			Name: j.Name,
+			TTL:  time.Duration(j.TTL) * time.Second,
+		}
+		records = append(records, tmp)
+	}
+
+	return records, nil
+}
+
+func (p *Provider) getRecordID(recordName string, zone string) (string, error) {
+	recordName = recordName + zone
+	listOpts := recordsets.ListOpts{
+		Type: "TXT",
+	}
+
+	allPages, err := recordsets.ListByZone(p.dnsClient, p.zoneID, listOpts).AllPages()
+	if err != nil {
+		return "", err
+	}
+
+	allRecordSets, err := recordsets.ExtractRecordSets(allPages)
+	if err != nil {
+		return "", err
+	}
+
+	for _, rr := range allRecordSets {
+		if recordName == rr.Name {
+			return rr.ID, nil
+		}
+	}
+
+	return "", nil
+}
+
+func (p *Provider) setRecordID(recordID string) {
+	p.recordID = recordID
```

Comment:
I'm pretty sure this is a race condition

Reply:
I removed it completely.